File: backend/src/main/java/com/yangshengzhou/lucky_sms/service/UserService.java
package com.yangshengzhou.lucky_sms.service;

import com.yangshengzhou.lucky_sms.mapper.UserMapper;
import com.yangshengzhou.lucky_sms.utils.JwtUtil;
import com.yangshengzhou.lucky_sms.vo.user.LoginVO;
import org.springframework.stereotype.Service;
import jakarta.annotation.Resource;

@Service
public class UserService {

    @Resource
    private UserMapper userMapper;

    @Resource
    private JwtUtil jwtUtil;

    public LoginVO loginByPhone(String phone) {
        LoginVO loginVO = userMapper.loginByPhone(phone);

        if (loginVO == null) {
            int rowNum = userMapper.registerByPhone(phone);
            if (rowNum == 0) {
                throw new RuntimeException("注册失败");
            }
        }
        loginVO = userMapper.loginByPhone(phone);

        String token = jwtUtil.generateToken(loginVO.getUid());
        loginVO.setToken(token);

        loginVO.setPhone(desensitizePhone(loginVO.getPhone()));

        return loginVO;
    }

    public boolean resetPassword(String phone, String password) {
        int rowNum = userMapper.resetPasswordByPhone(phone, password);
        return rowNum != 0;
    }

    public LoginVO loginByPassword(String phone, String password) {
        LoginVO loginVO = userMapper.loginByPassword(phone, password);

        if (loginVO == null) {
            throw new RuntimeException("用户名或密码错误");
        }

        String token = jwtUtil.generateToken(loginVO.getUid());
        loginVO.setToken(token);

        loginVO.setPhone(desensitizePhone(loginVO.getPhone()));

        return loginVO;
    }

    private String desensitizePhone(String phone) {
        if (phone == null || phone.length() != 11) {
            return phone;
        }
        return phone.substring(0, 3) + "****" + phone.substring(7);
    }
}

File: backend-go/service/user_service.go
package service

import (
	"errors"
	"lucky-sms/repository"
	"lucky-sms/utils"
	"lucky-sms/vo"
)

type UserService struct {
	userRepo *repository.UserRepository
}

func NewUserService() *UserService {
	return &UserService{
		userRepo: repository.NewUserRepository(),
	}
}

func (s *UserService) LoginByPhone(phone string) (*vo.LoginVO, error) {
	loginVO, err := s.userRepo.LoginByPhone(phone)
	if err != nil {
		rowsAffected, regErr := s.userRepo.RegisterByPhone(phone)
		if regErr != nil || rowsAffected == 0 {
			return nil, errors.New("注册失败")
		}
		
		loginVO, err = s.userRepo.LoginByPhone(phone)
		if err != nil {
			return nil, errors.New("用户注册成功但查询失败")
		}
	}

	token, err := utils.GenerateToken(loginVO.UID)
	if err != nil {
		return nil, err
	}

	loginVO.Token = token
	loginVO.Phone = utils.DesensitizePhone(loginVO.Phone)

	return loginVO, nil
}

func (s *UserService) LoginByPassword(phone, password string) (*vo.LoginVO, error) {
	loginVO, err := s.userRepo.LoginByPassword(phone, password)
	if err != nil {
		return nil, errors.New("用户名或密码错误")
	}

	token, err := utils.GenerateToken(loginVO.UID)
	if err != nil {
		return nil, err
	}

	loginVO.Token = token
	loginVO.Phone = utils.DesensitizePhone(loginVO.Phone)

	return loginVO, nil
}

func (s *UserService) ResetPassword(phone, password string) (*vo.ResetPasswordVO, error) {
	rowsAffected, err := s.userRepo.ResetPasswordByPhone(phone, password)
	if err != nil || rowsAffected == 0 {
		return &vo.ResetPasswordVO{Success: false}, errors.New("重置密码失败")
	}
	return &vo.ResetPasswordVO{Success: true}, nil
}

func (s *UserService) GetUserByID(id int) (*vo.UserManagementVO, error) {
	return nil, nil
}
